server/src/main/java/com/google/fhir/gateway/interfaces/AccessDecision.java

```diff
@@ -48,10 +48,11 @@ public interface AccessDecision {
    * <p>An example of this is when a new patient is created as the result of the query and that
    * patient ID should be added to some access lists.
    *
+   * @param request the client to server request details
    * @param response the response returned from the FHIR store
    * @return the response entity content (with any post-processing modifications needed) if this
    *     reads the response; otherwise null. Note that we should try to avoid reading the whole
    *     content in memory whenever it is not needed for post-processing.
    */
-  String postProcess(HttpResponse response) throws IOException;
+  String postProcess(RequestDetailsReader request, HttpResponse response) throws IOException;
```

Comment:
I think you need to update AccessGrantedAndUpdateList as well ([here](https://github.com/google/fhir-gateway/blob/850f0ee82afff52102b4656d9ebe5f61eccec736/plugins/src/main/java/com/google/fhir/gateway/plugin/AccessGrantedAndUpdateList.java#L78)).

Reply:
Thanks, done!